vispy/scene/visuals.py

```diff
@@ -71,7 +71,10 @@ def picking(self, p):
             return
         self._picking = p
         self._picking_filter.enabled = p
-        self.update_gl_state(blend=not p)
+        if p:
+            self.push_gl_state_update(blend=False)
+        else:
+            self.pop_gl_state()
```

Comment:
This assumes that `.picking` is being set and then unset, right? What if someone sets it and then never unsets it. Does reverting this back to what it was make the issues you were running into come back?

Reply:
The idea of removing the gl state modifications from here is intriguing, but yeah the backwards incompatibility is a bit too much.

I guess my worry here is something of an edge case so I'm OK leaving this as-is. The user would have to be doing something in a not very context manager-y way. I guess the biggest issue would be if someone did `vis.picking = False` multiple times, but maybe that's just something that we deal with when the bug reports come in (if any).

Side note: Just saw on your profile you went to UW-Madison. Me too and still working there. Small world.

Reply:
Oh cool - I grew up in Madison and attended UW for both undergrad and grad school. Madison is a great place and I loved my time there. I left in 2014 which I'm now realizing is almost 10 years ago!